server/auth_function/lambda_function.py

```diff
@@ -82,9 +83,13 @@ def lambda_handler(event: event_type.Event, context: Any) -> event_type.Event:
         audit_event_log["requestingUser"]["userType"] = USER_TYPE_CODE_DICT[
             event["request"]["userAttributes"]["custom:idp_name"]
         ]
-        audit_event_log["requestingUser"]["idpUserName"] = event["request"][
-            "userAttributes"
-        ]["custom:idp_username"]
+        if (
+            audit_event_log["requestingUser"]["userType"] == USER_TYPE_IDIR
+            or audit_event_log["requestingUser"]["userType"] == USER_TYPE_BCEID_BUSINESS
+        ):
+            audit_event_log["requestingUser"]["idpUserName"] = event["request"][
+                "userAttributes"
+            ]["custom:idp_username"]
```

Comment:
This should fix the issue, but doesn't actually handle logging of the BCSC identity as part of the audit record. So if you want to deploy this fix quickly, a subsequent ticket should be raised to address logging of BCSC identity.

Reply:
yeah, I think bcsc doesn't have the idp_username, but we might could show the display name if that makes sense. I'll verify what we can get and include in the follow up ticket #1018.